source/data-formats/modeling-bson-data.txt

```diff
@@ -35,184 +127,183 @@ default:
-           $this->name = $data['name'];
-           $this->createdAt = $data['createdAt'];
-       }
+     "_id" : ObjectId("56fad2c36118fd2e9820cfc1"),
+     "__pclass" : BinData(128,"UGVyc29u"),
+     "name" : "Bob",
+     "createdAt" : ISODate("2016-03-29T19:08:51.218Z")
    }
 
-The following example constructs a ``Person`` object, inserts it into the
-database, and reads it back as an object of the same type:
+The {+library-short+} automatically adds the ``__pclass`` field to keep
+track of the document's corresponding class name, which allows you to
+deserialize the document into a ``Person`` object.
 
-.. code-block:: php
+.. note::
 
-   <?php
+   You can use the ``Persistable`` interface for root and embedded BSON documents
+   only, not BSON arrays.
 
-   $collection = (new MongoDB\Client)->test->persons;
+.. _php-enums:
 
-   $result = $collection->insertOne(new Person('Bob'));
+Enum Values
+~~~~~~~~~~~
 
-   $person = $collection->findOne(['_id' => $result->getInsertedId()]);
+You can serialize and deserialize backed enums into BSON data. Backed 
+enum values serialize as their case value, while pure enums without
+case values cannot be directly serialized. To perform these conversions,
+you must specify serialization logic by defining the ``bsonSerialize()``
+and ``bsonUnserialize()`` methods in your class definition.
 
-   var_dump($person);
+.. tip::
 
-The output would then resemble:
+   To learn more about backed enums, see :php:`Backed enums <enumerations.backed>`
+   in the {+extension-short+} documentation.
 
-.. code-block:: none
+Example
+```````
 
-   object(Person)#18 (3) {
-     ["id":"Person":private]=>
-     object(MongoDB\BSON\ObjectId)#15 (1) {
-       ["oid"]=>
-       string(24) "56fad2c36118fd2e9820cfc1"
-     }
-     ["name":"Person":private]=>
-     string(3) "Bob"
-     ["createdAt":"Person":private]=>
-     object(MongoDB\BSON\UTCDateTime)#17 (1) {
-       ["milliseconds"]=>
-       int(1459278531218)
-     }
-   }
+Consider the following ``User`` class definition, which specifies
+logic for serializing and deserializing its fields into BSON values.
+The class includes a ``role`` field, which has a backed enum value:
 
-The same document in the MongoDB shell might display as:
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :emphasize-lines: 15, 24
+    :dedent:
+    :start-after: start-backed-enum
+    :end-before: end-backed-enum
 
-.. code-block:: js
+The following example constructs a ``User`` object with a ``role`` field,
+inserts it into the database, and reads it back as an object of the same type:
 
-   {
-     "_id" : ObjectId("56fad2c36118fd2e9820cfc1"),
-     "__pclass" : BinData(128,"UGVyc29u"),
-     "name" : "Bob",
-     "createdAt" : ISODate("2016-03-29T19:08:51.218Z")
-   }
+.. io-code-block::
+   :copyable:
 
-.. note::
+   .. input:: /includes/bson.php
+      :start-after: start-enum-serialize
+      :end-before: end-enum-serialize
+      :language: php
+      :dedent:
```

Comment:
S: maybe its worth adding a sentence below here that says that the USER enum value was stored as a 1 in MongoDB, but deserialized correctly into the enum value. But can also leave out if its obvious/redundant.